tls/_constructs.py

```diff
@@ -74,12 +72,12 @@
 
 ServerName = Struct(
     "server_name",
-    EnumClass(UBInt8("name_type"), enums.NameType),
-    Switch(
-        "name",
-        lambda ctx: ctx.name_type,
-        {
-            enums.NameType.HOST_NAME: HostName
+    *EnumSwitch(
```

Comment:
It's believable that `EnumSwitch` just works here, because it's equivalent to `EnumClass` followed by a `Switch` statement.